SimCalorimetry/HGCalAssociatorProducers/plugins/LCToSCAssociatorByEnergyScoreImpl.cc

```diff
@@ -373,8 +379,17 @@ ticl::association LCToSCAssociatorByEnergyScoreImpl<HIT>::makeConnections(
     // It is the inverse of the denominator of the LCToSC score formula. Observe that this is the sum of the squares.
     float invLayerClusterEnergyWeight = 0.f;
     for (auto const& haf : hits_and_fractions) {
-      const HIT* hit = hits_[hitMap_->at(haf.first)];
-      invLayerClusterEnergyWeight += (haf.second * hit->energy()) * (haf.second * hit->energy());
+      if constexpr (std::is_same_v<HIT, HGCRecHit>) {
```

Comment:
Since this pattern is everywhere, I'm wondering if it is a symptom of some bug? Or maybe it is there to debug a problem that has been fixed in the meantime?